File: content/panorama/scripts/custom_game/shared/AddonInfo.js
// package: com.isycat.dotaaddon.shared

const AddonInfo = Object.freeze({
    NAME: "ktoxtest",
    VERSION: "1.0.0",
    getWelcomeMessage() {
        return `Hello from ${this.NAME} v${this.VERSION}!`;
    },
    printSomeNumbers() {
        KtoxSequence.of([1, 2, 3]).forEach((it) => {
            console.log(it);
        });
        [1, 2, 3].forEach((it) => {
            console.log(it);
        });
        KtoxSequence.of([1, 2, 3]).forEach((it) => {
            console.log(it);
        });
    },
});

File: shared/src/main/kotlin/com/isycat/dotaaddon/shared/AddonInfo.kt
package com.isycat.dotaaddon.shared

/**
 * Shared constants used by both the Lua and Panorama modules. The `com.isycat.ktox.shared` plugin treats
 * every file in this module as a library source, so no per-file `@KtoxLibrarySource` is needed.
 */
object AddonInfo {
    const val NAME = "ktoxtest"
    const val VERSION = "1.0.0"

    fun getWelcomeMessage(): String = "Hello from $NAME v$VERSION!"
}
